src/Views/Components/ListTable/CustomFilter/useAsyncCampaigns.ts

```diff
@@ -0,0 +1,86 @@
+import { useCallback, useEffect, useState } from 'react';
+import apiFetch from '@wordpress/api-fetch';
+import { createCampaignQueryParams, processOptionsForMenu, formatCampaignOptions } from './utils';
+import { Campaign, CampaignOption } from './utils';
+
+const CAMPAIGNS_PER_PAGE = 30;
+
+type UseCampaignAsyncSelectReturn = {
+    selectedOption: CampaignOption | null;
+    setSelectedOption: (option: CampaignOption | null) => void;
+    loadOptions: (searchInput: string) => Promise<{
+        options: CampaignOption[];
+        hasMore: boolean;
+    }>;
+    mapOptionsForMenu: (options: CampaignOption[]) => CampaignOption[];
+    error: Error | null;
+}
+
+/**
+ * Custom hook for handling async form selection with pagination and search
+ *
+ * @unreleased
+ */
+export function useCampaignAsyncSelect(): UseCampaignAsyncSelectReturn {
+    const [page, setPage] = useState(0);
+    const [selectedOption, setSelectedOption] = useState<CampaignOption | null>(null);
+    const [error, setError] = useState<Error | null>(null);
+
+    // Load options function for AsyncPaginate
+    const loadOptions = useCallback(async (search: string) => {
+        const currentPage = search ? 1 : page + 1;
+
+        setError(null);
+
+        try {
+            const queryParams = createCampaignQueryParams({
+                perPage: CAMPAIGNS_PER_PAGE,
+                page: currentPage,
+                search: search || undefined,
+            });
+
+            const campaigns = await apiFetch<Campaign[]>({
+                path: `/givewp/v3/campaigns?${queryParams.toString()}`,
+            });
```

Comment:
@JoshuaHungDinh Is there any reason we're not using the campaigns entity here?

Example: https://github.com/impress-org/givewp/blob/a6165c0acca3e0e51d8ae96582bd7d6aa5dd82f6/src/Campaigns/Blocks/shared/hooks/useCampaigns.ts#L23

Reply:
Okay, I think it's worth exploring if entities can be used here instead of the manual fetch.  I'm okay refactoring this part later so it's not a blocker.